keps/sig-node/4603-tune-crashloopbackoff/README.md

```diff
@@ -0,0 +1,1431 @@
+-->
+
+Yes, disable is supported.
+
+For `ReduceDefaultCrashLoopBackoffDecay`, if this is disabled, once kubelet is
+restarted it will initialize the default backoff to the prior initial value of
+10s, and all restart delays thereafter will be calculated against this equation.
+
+For `EnableRapidCrashLoopBackoffDecay`, if this is disabled, once kube-apiserver
+is restarted it will serve `restartPolicy` fields set to `Rapid` as `Always`.
+
+###### What happens if we reenable the feature if it was previously rolled back?
+
+Both features can also be reenabled.
+
+For `ReduceDefaultCrashLoopBackoffDecay`, if this is reenabled, once kubelet is
+restarted it will initialize the default backoff again to the new initial value
+of 1s, and all restart delays thereafter will be calculated against this
+equation.
+
+For `EnableRapidCrashLoopBackoffDecay`, if this is disabled, once kube-apiserver
+is restarted it will serve `restartPolicy` fields set to `Rapid` again as
+`Rapid`, which kubelet will be able to interpret.
+
+###### Are there any tests for feature enablement/disablement?
+
+<!--
+The e2e framework does not currently support enabling or disabling feature
+gates. However, unit tests in each component dealing with managing data, created
+with and without the feature, are necessary. At the very least, think about
+conversion tests if API types are being modified.
+
+Additionally, for features that are introducing a new API field, unit tests that
+are exercising the `switch` of feature gate itself (what happens if I disable a
+feature gate after having objects written with the new field) are also critical.
+You can take a look at one potential example of such test in:
+https://github.com/kubernetes/kubernetes/pull/97058/files#diff-7826f7adbc1996a05ab52e3f5f02429e94b68ce6bce0dc534d1be636154fded3R246-R282
+-->
+
+Yes, this requires tests for
+* switching `ReduceDefaultCrashLoopBackoffDecay` on or off
+* switching `EnableRapidCrashLoopBackoffDecay` off when there are workloads with
+  `restartPolicy: Rapid` set
+* switching `EnableRapidCrashLoopBackoffDecay` off when there are no workloads
+  with `restartPolicy: Rapid` set
+* switching `EnableRapidCrashLoopBackoffDecay` off, and then back on again, when
+  there are workloads with `restartPolicy: Rapid` set
+
+### Rollout, Upgrade and Rollback Planning
+
+<!--
+This section must be completed when targeting beta to a release.
+-->
+
+###### How can a rollout or rollback fail? Can it impact already running workloads?
+
+<!--
+Try to be as paranoid as possible - e.g., what if some components will restart
+mid-rollout?
+
+Be sure to consider highly-available clusters, where, for example,
+feature flags will be enabled on some API servers and not others during the
+rollout. Similarly, consider large clusters and how enablement/disablement
+will rollout across nodes.
+-->
+
+###### What specific metrics should inform a rollback?
+
+<!--
```

Comment:
You did mention metrics in the alpha stage, can you list the ones you're thinking of here? They will be required for beta promotion.

Reply:
Added in [b1feb73](https://github.com/kubernetes/enhancements/pull/4604/commits/b1feb73f831c26a1410f1c7b9ba8bb4295a5c3de) regarding rollback in particular